components/org.wso2.identity.apps.taglibs.layout.controller/src/main/java/org/wso2/identity/apps/taglibs/layout/controller/compiler/parsers/DefaultParser.java

```diff
@@ -0,0 +1,228 @@
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.ComponentIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.ConditionIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.DataIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.DefaultIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.ExecutableIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.NoIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.identifiers.NotConditionIdentifier;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.resolvers.FileResolver;
+import org.wso2.identity.apps.taglibs.layout.controller.compiler.resolvers.Resolver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The default parser, use to compile the layout file.
+ */
+public class DefaultParser implements Parser {
+
+    private final Pattern pattern = Pattern.compile(
+            "\\{\\{([-_a-zA-Z0-9]+)\\}\\}|" + "\\{\\{\\{([-_a-zA-Z0-9]+)\\}\\}\\}|" + "\\{\\{#([-_a-zA-Z0-9]+)\\}\\}|" +
+                    "\\{\\{/([-_a-zA-Z0-9]+)\\}\\}|" + "\\{\\{\\^([-_a-zA-Z0-9]+)\\}\\}");
+    private final Resolver resolver;
+
+    /**
+     * Constructor.
+     */
+    public DefaultParser() {
+
+        resolver = new FileResolver();
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resolver File resolver to read the layout file.
+     */
+    public DefaultParser(Resolver resolver) {
+
+        this.resolver = resolver;
+    }
+
+    /**
+     * Execute the layout file and create a compiled layout file as an object.
+     *
+     * @param file File path for the layout as a URL object.
+     * @return Compiled layout file.
+     */
+    @Override
+    public ExecutableIdentifier compile(URL file) {
+
+        BufferedReader reader = (BufferedReader) resolver.getReader(file);
+        CompileContext context = new CompileContext();
+        ExecutableIdentifier temp = compile(reader, context, null);
+        resolver.closeResources();
+        return temp;
+    }
+
+    /**
+     * Recursive compiling function to compile the layout file.
+     *
+     * @param reader         BufferedReader to read the layout file.
+     * @param context        Compiling context.
+     * @param identifierName Name of the recently identified block identifier
+     *                       (Ex:- ConditionIdentifier, NotConditionIdentifier).
+     * @return Compiled layout file or block of identifiers.
+     */
+    private ExecutableIdentifier compile(BufferedReader reader, CompileContext context, String identifierName) {
+
+        ArrayList<ExecutableIdentifier> allIdentifiers = new ArrayList<ExecutableIdentifier>();
```

Comment:
```suggestion
        ArrayList<ExecutableIdentifier> allIdentifiers = new ArrayList<>();
```
Explicit type argument ExecutableIdentifier can be replaced with <>